File: backend/app/schemas/order.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class OrderItemResponse(BaseModel):
    variant_name: str
    product_name: str
    quantity: int
    unit_price: Decimal
    line_total: Decimal
    provider_sku: str | None = None

    model_config = {"from_attributes": True}


class OrderResponse(BaseModel):
    id: uuid.UUID
    campaign_id: uuid.UUID | None = None
    campaign_title: str | None = None
    campaign_slug: str | None = None
    customer_email: str
    customer_name: str | None = None
    subtotal: Decimal
    tip_amount: Decimal
    total: Decimal
    status: str
    fulfillment_provider: str | None = None
    external_order_id: str | None = None
    fulfillment_status: str | None = None
    fulfillment_error: str | None = None
    tracking_number: str | None = None
    fulfillment_submitted_at: datetime | None = None
    items: list[OrderItemResponse]
    created_at: datetime
    facebook_pixel_id: str | None = None
    google_analytics_id: str | None = None

    model_config = {"from_attributes": True}


class OrderListResponse(BaseModel):
    orders: list[OrderResponse]
    total: int

File: frontend/src/lib/types/order.ts
export interface OrderItem {
  variant_name: string;
  product_name: string;
  quantity: number;
  unit_price: number;
  line_total: number;
}

export interface Order {
  id: string;
  campaign_id: string | null;
  campaign_title: string | null;
  campaign_slug?: string | null;
  customer_email: string;
  customer_name: string | null;
  subtotal: number;
  tip_amount: number;
  total: number;
  status: string;
  items: OrderItem[];
  created_at: string;
  facebook_pixel_id?: string | null;
  google_analytics_id?: string | null;
}

export interface CartItem {
  variant_id: string;
  variant_name: string;
  quantity: number;
  unit_price: number;
  line_total: number;
}

export interface Cart {
  id: string;
  campaign_id: string;
  campaign_slug: string;
  campaign_title: string;
  items: CartItem[];
  subtotal: number;
}
